### PR TITLE
chore(table demos): update filterable demo from deprecated to new table

### LINKED ISSUE #9030: Table  - Filterable demo cleanup
Update the demo to use the Table (formerly TableComposable) and not the deprecated Table components.

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #9030

### CODE REVIEW

**Review by nicolethoen — COMMENTED** [review]
I see a few things we could clean up while we are tinkering with this demo:

Looks like you can remove the following unused dependencies:

```
Dropdown,
DropdownItem,
DropdownPosition,
DropdownToggle,
```

`minWidth` is being passed twice to the Select on line 1376
`filters` is not ever used - defined on line 1399

**Review by jenny-s51 — COMMENTED** [review]
Thanks @nicolethoen ! updated

**Review by nicolethoen — COMMENTED** [review]
I think MenuToggle and MenuToggleElement need to be imported by your demo.

**Review by jenny-s51 — COMMENTED** [review]
Thank you @nicolethoen - added the missing imports. I'll be double-checking any demo PRs in codesandbox from now on 🙂 

Good find on the import path - assigned myself to https://github.com/patternfly/patternfly-org/issues/3525, will be looking into it

File: packages/react-table/src/docs/demos/Table.md [hunk 1]
```diff
@@ -1330,13 +1299,13 @@ class FilterTableDemo extends React.Component {
...
-      <SelectOption
-        hasCheckbox
-        key="statusMaint"
-        itemId="Needs Maintainence"
-        isSelected={filters.status.includes("Needs Maintainence")}
+      <SelectOption 
+        hasCheckbox 
+        key="statusMaint" 
+        itemId="Needs Maintenance"
+        isSelected={filters.status.includes("Needs Maintenance")}
```
Comment (tlabaj) [inline packages/react-table/src/docs/demos/Table.md:1306]: nit. sentence case here and line 1308

**Comment by nicolethoen** [comment]
> Good find on the import path - assigned myself to [patternfly/patternfly-org#3525](https://github.com/patternfly/patternfly-org/issues/3525), will be looking into it

I think @evwilkin has done work on updating relative imports in codesandbox imports in the past. So once he's back, he may have good insights here

**Review by tlabaj — APPROVED** [review]
lgtm

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**